src/main/java/Handler/FileHandler.java

```diff
@@ -0,0 +1,73 @@
+import java.io.BufferedReader; 
+import java.io.File; 
+import java.io.FileReader; 
+import java.io.IOException; 
+ 
+public class FileHandler implements Handler { 
+  String directoryPath; 
+  String filePath; 
+  Request request; 
+ 
+  public FileHandler(String directoryPath) throws NonexistentDirectoryException { 
+    File directory = new File(directoryPath);
+    if (directory.exists()) {
+      this.directoryPath = directoryPath; 
+    } else {
+      String errorMessage = this.directoryPath + " does not exist. Enter a valid directory.";
+      throw new NonexistentDirectoryException(errorMessage);
+    }
+  } 
+   
+  public Response generateResponse(Request request) { 
+    this.request = request; 
```

Comment:
The `createFileFromURI()` helper method calls `request.getURI()`. The method creates an instance of `File` by appending the request's URI to the directory path passed in to the `FileHandler` constructor. Since I don't use the request object anywhere else, should I remove `request` as an instance variable and replace it with `uri` instead?